sdk/tables/azure-data-tables/tests/test_table_batch_cosmos.py

```diff
@@ -178,173 +182,158 @@ def test_inferred_types(self):
-            entity = Entity()
+            entity = TableEntity()
             entity.PartitionKey = '001'
             entity.RowKey = 'batch_merge'
             entity.test2 = 'value1'
 
             batch = self.table.create_batch()
-            batch.update_item(entity, mode='MERGE')
-            resp = self.table.commit_batch(batch)
+            batch.update_entity(entity, mode='MERGE')
+            resp = self.table.send_batch(batch)
 
             # Assert
             self.assertIsNotNone(resp)
-            entity, headers = self.table.read_item('001', 'batch_merge', response_hook=lambda e, h: (e, h))
+            entity, headers = self.table.get_entity('001', 'batch_merge', response_hook=lambda e, h: (e, h))
             self.assertEqual('value1', entity.test2)
             self.assertEqual(1234567890, entity.test4)
             self.assertEqual(list(resp)[0].headers['Etag'], headers['etag'])
         finally:
             self._tear_down()
 
-    @pytest.mark.skip("pending")
+    @pytest.mark.skipif(sys.version_info < (3, 0), reason="requires Python3")
     @CachedResourceGroupPreparer(name_prefix="tablestest")
     @CachedCosmosAccountPreparer(name_prefix="tablestest")
     def test_batch_update_if_match(self, resource_group, location, cosmos_account, cosmos_account_key):
         # Arrange
         self._set_up(cosmos_account, cosmos_account_key)
         try:
             entity = self._create_random_entity_dict()
-            etag = self.table.create_item(entity, response_hook=lambda e, h: h['etag'])
+            resp = self.table.create_entity(entity=entity)
+            etag = resp['etag']
 
             # Act
             sent_entity = self._create_updated_entity_dict(entity['PartitionKey'], entity['RowKey'])
             batch = self.table.create_batch()
-            batch.update_item(sent_entity, etag=etag, match_condition=MatchConditions.IfNotModified)
-            resp = self.table.commit_batch(batch)
+            batch.update_entity(
+                sent_entity,
+                etag=etag,
+                match_condition=MatchConditions.IfNotModified,
+                mode=UpdateMode.REPLACE
+            )
+            transaction_result = self.table.send_batch(batch)
 
             # Assert
-            self.assertIsNotNone(resp)
-            entity, headers = self.table.read_item(entity['PartitionKey'], entity['RowKey'], response_hook=lambda e, h: (e, h))
+            self._assert_valid_batch_transaction(transaction_result, 1)
+            self.assertIsNotNone(transaction_result.get_entity(sent_entity['RowKey']))
+
+            entity = self.table.get_entity(partition_key=entity['PartitionKey'], row_key=entity['RowKey'])
             self._assert_updated_entity(entity)
-            self.assertEqual(list(resp)[0].headers['Etag'], headers['etag'])
         finally:
             self._tear_down()
 
-    @pytest.mark.skip("pending")
+    @pytest.mark.skipif(sys.version_info < (3, 0), reason="requires Python3")
```

Comment:
why does this test require python3

Reply:
Batching as a whole requires Python3 as a whole